pkg/controller/registry/resolver/util_test.go

```diff
@@ -307,6 +307,16 @@ func apiSetToProperties(crds, apis APISet) (out []*api.Property) {
 			Value: string(val),
 		})
 	}
+	if deprecated {
+		val, err := json.Marshal(opregistry.DeprecatedProperty{})
+		if err != nil {
+			panic(err)
```

Comment:
`Panic` seems drastic. Can we just return the error instead?

Reply:
I'm staying consistent with the pattern https://github.com/operator-framework/operator-lifecycle-manager/pull/1673/files/4988c89d50d10f84b1d2d90df81594cc5121db37#diff-6c0a163d01fd638f4e9b516fee8ad852L302

Reply:
Not sure what that link is pointing towards, but it's bad to panic from inside a running pod.

Reply:
Maybe not ideal, but it is test code. Reading into the intention behind the existing panics, I'm guessing that the choice was between panicking here or changing the function's signature, and thereby making it unusable from within struct literals.

Big :+1: from me in general in investing in test code, but it may not need to block this PR. I don't think changing the function signature is strictly an improvement, so it would take some time to find a way to handle errors gracefully without introducing more boilerplate at every call site.

Reply:
IMHO it completely slipped me that this is a test. This is fine.